### PR TITLE
new CLA assistant

### PR DESCRIPTION
**Pull request by bosconi**
Switch to using [the GitHub Actions based version](https://github.com/contributor-assistant/github-action) of the CLA Assistant that we have been using.

### Motivation

The [hosted version of the CLA Assistant](https://cla-assistant.io/) has been flaky recently. We will use the Actions version instead, configured to write the signatures into a new public repo: https://github.com/MaterializeInc/cla

Once this PR is pushed we will see if we can merge the existing signatures into the signature file. 

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: .github/workflows/cla.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: "CLA Assistant"
+on:
+  issue_comment:
+    types: [created]
+  pull_request_target:
+    types: [opened,closed,synchronize]
+
+# explicitly configure permissions, in case your GITHUB_TOKEN workflow permissions are set to read-only in repository settings
```
Comment (benesch) [inline .github/workflows/cla.yml:8]:
```suggestion
```

File: .github/workflows/cla.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+permissions:
+  actions: write
+  contents: read
+  pull-requests: write
+  statuses: write
+
+jobs:
+  CLAAssistant:
```
Comment (benesch) [inline .github/workflows/cla.yml:16]:
```suggestion
  cla-assistant:
```

File: .github/workflows/cla.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        uses: contributor-assistant/github-action@v2.4.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PERSONAL_ACCESS_TOKEN: ${{ secrets.CLA_TOKEN }}
+        with:
+          path-to-signatures: 'signatures/version1/signatures.json'
+          path-to-document: 'https://gist.github.com/af3262c2eb70b14b8e96260e8acfc2cf' # CLA document
+          # branch should not be protected
+          branch: 'main'
+          allowlist: “materialize-bot,materialize-bot-monitoring,dependabot[bot]”
```
Comment (benesch) [inline .github/workflows/cla.yml:30]:
```suggestion
          allowlist: "materialize-bot,materialize-bot-monitoring,dependabot[bot]"
```

File: .github/workflows/cla.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          remote-organization-name: MaterializeInc
+          #remote-repository-name: enter the  remote repository name where the signatures should be stored (Default is storing the signatures in the same repository)
+          remote-repository-name: cla
+          #create-file-commit-message: 'For example: Creating file for storing CLA Signatures'
+          #signed-commit-message: 'For example: $contributorName has signed the CLA in $owner/$repo#$pullRequestNo'
+          #custom-notsigned-prcomment: 'pull request comment with Introductory message to ask new contributors to sign'
+          #custom-pr-sign-comment: 'The signature to be committed in order to sign the CLA'
+          #custom-allsigned-prcomment: 'pull request comment when all contributors has signed, defaults to **CLA Assistant Lite bot** All Contributors have signed the CLA.'
+          #lock-pullrequest-aftermerge: false - if you don't want this bot to automatically lock the pull request after merging (default - true)
+          #use-dco-flag: true - If you are using DCO instead of CLA
```
Comment (benesch) [inline .github/workflows/cla.yml:43]:
```suggestion
          # Store signatures in a central repository so they can be shared by
          # all repositories in the organization.
          remote-organization-name: MaterializeInc
          remote-repository-name: cla
```

File: .github/workflows/cla.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - name: "CLA Assistant"
+        if: (github.event.comment.body == 'recheck' || github.event.comment.body == 'I have read the CLA Document and I hereby sign the CLA') || github.event_name == 'pull_request_target'
+        uses: contributor-assistant/github-action@v2.4.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PERSONAL_ACCESS_TOKEN: ${{ secrets.CLA_TOKEN }}
+        with:
+          path-to-signatures: 'signatures/version1/signatures.json'
+          path-to-document: 'https://gist.github.com/af3262c2eb70b14b8e96260e8acfc2cf' # CLA document
```
Comment (benesch) [inline .github/workflows/cla.yml:27]: This is a raw text file that's hard to read—I propose we add a Markdown-formatted version in MaterializeInc/cla and then link to that nicely rendered version here.

**Review by benesch — COMMENTED** [review]
Sweet, thank you! I added a few suggestions to remove the explanatory comments, to keep things clean.

File: .github/workflows/cla.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+          # branch should not be protected
```
Comment (benesch) [inline .github/workflows/cla.yml:28]:
```suggestion
```

File: .github/workflows/cla.yml (hunk 1 above)
Comment (bosconi) [reply .github/workflows/cla.yml]: Done

File: .github/workflows/cla.yml (hunk 3 above)
Comment (bosconi) [reply .github/workflows/cla.yml]: Done.

File: .github/workflows/cla.yml (hunk 4 above)
Comment (bosconi) [reply .github/workflows/cla.yml]: Done

File: .github/workflows/cla.yml (hunk 6 above)
Comment (bosconi) [reply .github/workflows/cla.yml]: Done

File: .github/workflows/cla.yml (hunk 2 above)
Comment (bosconi) [reply .github/workflows/cla.yml]: Done

File: .github/workflows/cla.yml (hunk 5 above)
Comment (bosconi) [reply .github/workflows/cla.yml]: https://github.com/MaterializeInc/cla/blob/main/Materialize_CLA.md looks good as a markdown file with same content as the gist.

**Review by benesch — APPROVED** [review]

File: .github/workflows/cla.yml (hunk 3 above)
Comment (def-) [reply .github/workflows/cla.yml:55]: Doesn't seem to be done.

**Review by def- — COMMENTED** [review]
Forgot to push latest change?

File: .github/workflows/cla.yml (hunk 3 above)
Comment (nrainer-materialize) [reply .github/workflows/cla.yml:55]: It was done in https://github.com/MaterializeInc/materialize/pull/28682/commits/440245d28de37d9f3811b1a61b5e959e8e0a2be7 but reverted for whatever reason in https://github.com/MaterializeInc/materialize/pull/28682/commits/03c29dcf91caa1a16b202a5705a94192b05a51ea.

**Pull request merged**